src/qibo/models/circuit.py

```diff
@@ -1265,11 +1265,14 @@ def read_args(args):
         gate_list = (
             []
         )  # List[Tuple[str, List[int]]]: List of (gate name, list of target qubit ids)
-        for line in lines:
-            command, args = line.split(None, 1)
-            # remove spaces
-            command = command.replace(" ", "")
-            args = args.replace(" ", "")
+        composite_gates = {}  # composite gates created with the "gate" command
+
+        def parse_line(line):
+            line = line.replace(r"/\s\s+/g", " ").replace(r"^[\s]+", "")
```

Comment:
```suggestion
            line = line.replace(r"/\s{2,}/g", " ").replace(r"^[\s]+", "")
```

But I really believe we should avoid building a parser procedurally with regexes...